components/tikv_util/src/resource_control.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+/// This mod provide some utility types and functions for resource control.
+use std::borrow::Cow;
+
+use kvproto::kvrpcpb::ResourceControlContext;
+use strum::{EnumCount, EnumIter};
+
+/// default resource group name
+pub const DEFAULT_RESOURCE_GROUP_NAME: &str = "default";
+
+const OVERRIDE_PRIORITY_MASK: u8 = 0b1000_0000;
+const RESOURCE_GROUP_NAME_MASK: u8 = 0b0100_0000;
+
+#[derive(Clone, Default)]
+pub struct TaskMetadata<'a> {
```

Comment:
why move to `tikv_util` instead of `resource_control/util`

Reply:
Because yatp_pool under `tikv_util` depends on this type to extract priority from `Extras`, move these common types under tike_util can let us handle dependency easier.